libs/constrained-generators/src/Constrained/Syntax.hs

```diff
@@ -0,0 +1,25 @@
+{-# LANGUAGE ImportQualifiedPost #-}
+
+module Constrained.Syntax where
+
+import Language.Haskell.TH qualified as TH
+import Language.Haskell.TH.Quote qualified as TH
+
+mkNamed :: String -> TH.Q TH.Pat
+mkNamed x =
+  pure $
+    TH.ViewP (TH.AppE (TH.VarE $ TH.mkName "name") (TH.LitE $ TH.StringL x)) (TH.VarP $ TH.mkName x)
+
+mkNamedExpr :: String -> TH.Q TH.Exp
+mkNamedExpr x =
+  pure $
+    TH.AppE (TH.AppE (TH.VarE $ TH.mkName "name") (TH.LitE $ TH.StringL x)) (TH.VarE $ TH.mkName x)
+
+var :: TH.QuasiQuoter
+var =
+  TH.QuasiQuoter
+    { TH.quotePat = mkNamed . head . words
+    , TH.quoteExp = mkNamedExpr . head . words
+    , TH.quoteDec = const $ fail "var should only be used at binding sites and in expressions"
+    , TH.quoteType = const $ fail "var should only be used at binding sites and in expressions"
+    }
```

Comment:
Maybe a comment about how this relates to quasiquoted  variable in a lambda expression.

spec = constrained $ \ [var| abc |] -> ...

that in    [xxx| abc |]  the system looks for a Quasiquoter with name "xxx", and uses it to parse the string "abc" into internal Haskell syntax.

Reply:
I've added a clearer explanation